Install fails with npm 7.9.0 (Unsupported URL Type "link:")

How to reproduce:

1. Install npm 7.9.0 (works fine on npm 6.14)
2. npm i pretter/prettier
3. exits with:

```
npm ERR! npm ERR! code EUNSUPPORTEDPROTOCOL
npm ERR! npm ERR! Unsupported URL Type "link:": link:scripts/tools/eslint-plugin-prettier-internal-rules
npm ERR!
```

It seems like npm breaks on a change in the dev dependencies. It could be a bug in npm since it shouldn't care about the dev dependencies. This is the commit that introduces the error:

https://github.com/prettier/prettier/blob/1b2c72df101208310bc22964ac06766a5453f5d5/package.json#L113

I've tried just switching back to `file:` but then eslint gives errors about node comments:

```
prettier/src/language-js/parser-meriyah.js
  56:7  error  Do not access node.comments  prettier-internal-rules/no-node-comments

prettier/src/language-js/parser/json.js
  23:47  error  Do not access node.comments  prettier-internal-rules/no-node-comments
  25:33  error  Do not access node.comments  prettier-internal-rules/no-node-comments

prettier/src/language-js/pragma.js
  14:20  error  Do not access node.comments  prettier-internal-rules/no-node-comments
  25:35  error  Do not access node.comments  prettier-internal-rules/no-node-comments

prettier/src/language-js/utils.js
  1284:38  error  Do not access node.comments  prettier-internal-rules/no-node-comments
  1288:22  error  Do not access node.comments  prettier-internal-rules/no-node-comments
  1298:36  error  Do not access node.comments  prettier-internal-rules/no-node-comments
  1302:22  error  Do not access node.comments  prettier-internal-rules/no-node-comments
  1302:51  error  Do not access node.comments  prettier-internal-rules/no-node-comments

✖ 10 problems (10 errors, 0 warnings)
```

It looks like this rule doesn't get loaded with just `file:`

```
  - files:
      - src/language-js/**/*.js
    rules:
      prettier-internal-rules/no-node-comments:
        - error
        - file: "src/language-js/utils.js"
          functions:
            - hasComment
            - getComments
        - "src/language-js/parse-postprocess.js"
        - "src/language-js/parser-babel.js"
        - "src/language-js/parser-meriyah.js"
        - "src/language-js/pragma.js"
        - "src/language-js/parser/json.js"
```

I'm not really sure how this eslint config is meant to work (should the rules be ignored or enforced in the files specified?), but I will gladly submit a PR if I could get some guidence :)

--- a/package.json
+++ b/package.json
@@ -110,7 +110,7 @@
     "eslint-formatter-friendly": "7.0.0",
     "eslint-plugin-import": "2.23.4",
     "eslint-plugin-jest": "24.3.6",
-    "eslint-plugin-prettier-internal-rules": "link:scripts/tools/eslint-plugin-prettier-internal-rules",
+    "eslint-plugin-prettier-internal-rules": "2.0.1",
     "eslint-plugin-react": "7.24.0",
     "eslint-plugin-regexp": "0.13.1",
     "eslint-plugin-unicorn": "34.0.1",

